### ISSUE TITLE
markdown-edit-code-block incorrectly indents code in the indirect buffer

### ISSUE THREAD

**Issue by phst** [jrblevin/markdown-mode#375]
## Expected Behavior

If a GFM block itself is indented, <kbd>C-c '</kbd> should not add indentation to the indirect buffer.

## Actual Behavior

<kbd>C-c '</kbd> adds indentation to such block.

## Steps to Reproduce

1. Create a test file such as

       $ cat /tmp/a.md 
       *   a
       
           ``` shell
           abc
           ```

1. 
       emacs -Q -l ../edit-indirect/edit-indirect.el -l markdown-mode.el /tmp/a.md
1. move point into the code block
2. <kbd>C-c '</kbd>
3. Code block will be indented by 4 spaces in the indirect buffer

## Software Versions

- Markdown Mode: markdown-mode, version 2.4-dev, Git version d18a8f856d19dfac8fa6e6e72b2448e262045fcc
- Emacs: GNU Emacs 26.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.22.24), modified by Debian
- OS: Debian testing

**Comment by syohex** [comment]
I suppose it is difficult to support this feature with current `edit-indirect` package specification/implementation.

**Comment by MyriaCore** [comment]
If `edit-indirect` mode interpreted rectangular regions properly, maybe we could specify a rectangular region whose width spans from the start of the codeblock to the end of the line? Writing a proposal for `edit-indirect` mode now.

**Comment by MyriaCore** [comment]
> Writing a proposal for `edit-indirect` mode now.

Just finished the proposal, it's over at Fanael/edit-indirect#17.

**Comment by Fanael** [comment]
You shouldn't need rectangular regions (they're a right mess anyway, and supporting them properly is hard) to do that. Instead, find out if the code block, including its start and end lines, is indented, and if it is, save the indentation level in a buffer local variable in the edit-indirect buffer. Then dedenting in the after creation hook and indenting back in the before commit hook should do the trick.

**Comment by MyriaCore** [comment]
I agree that that'd probably be a pretty OK solution. I might try and hack that into markdown-mode.el to see if i can implement that.

**Comment by MyriaCore** [comment]
Here's the current definition of `markdown-edit-code-block`:

https://github.com/jrblevin/markdown-mode/blob/cf6403186119cd3d25adc702845f969071060e20/markdown-mode.el#L8584-L8604

I ended up settling on this edit to save and set the indentation depth of the code block:

```elisp
(defun markdown-edit-code-block ()
  "Edit Markdown code block in an indirect buffer."
  (interactive)
  (save-excursion
    (if (fboundp 'edit-indirect-region)
        (let* ((bounds (markdown-get-enclosing-fenced-block-construct))
               (begin (and bounds (goto-char (nth 0 bounds)) (point-at-bol 2)))
               (end (and bounds (goto-char (nth 1 bounds)) (point-at-bol 1))))
          (if (and begin end)
              (let* ((indentation (and (goto-char (nth 0 bounds)) (current-indentation)))
                     (lang (markdown-code-block-lang))
                     (mode (or (and lang (markdown-get-lang-mode lang))
                               markdown-edit-code-block-default-mode))
                     (edit-indirect-guess-mode-function
                      (lambda (_parent-buffer _beg _end)
                        (funcall mode)))
                     (indirect-buf (edit-indirect-region begin end 'display-buffer)))
                (when (> indentation 0)
                  (with-current-buffer indirect-buf
                    (indent-rigidly (point-min) (point-max) (- indentation)))))
            (user-error "Not inside a GFM or tilde fenced code block")))
      (when (y-or-n-p "Package edit-indirect needed to edit code blocks. Install it now? ")
        (progn (package-refresh-contents)
               (package-install 'edit-indirect)
               (markdown-edit-code-block))))))
```

I'll note that this solution *won't* work for the following cases:

````
<!-- code block in list -->
- ```
  code
  ```
* ```
  code
  ```
1. ```
   code
   ```


<!-- in quote -->
> ```
> code
> ```
````

... but then again, code block detection was spotty for me in these cases *before*

**Comment by MyriaCore** [comment]
@syohex this issue should be closed, as #559 was merged via  [`09c9934^...2ac9be1`](https://github.com/jrblevin/markdown-mode/compare/09c99345a7ccdb6ac0a19fea5db3006ff08b19da%5E...2ac9be10d6d21801b7f4f130f3b4d82fd0ed7cfe)

**Comment by syohex** [comment]
Oh, sorry  I forgot. I have closed.
